packages/gatsby-plugin-static-image/README.md

```diff
@@ -0,0 +1,224 @@
+
+You can pass in options that match ones passed to the `ImageSharp` query:
+
+```js
+import React from "react"
+import { StaticImage as Img } from "gatsby-plugin-static-image"
+
+export const Dino = () => (
+  <Img
+    src="trex.png"
+    base64={false}
+    fluid
+    webP
+    grayscale
+    maxWidth={200}
+    alt="T-Rex"
+  />
+)
+```
+
+...is equivalent to:
+
+```js
+import React from "react"
+import Img from "gatsby-image"
+
+export const Dino = () => {
+  const data = useStaticQuery(graphql`
+    query LogoQuery {
+      file(relativePath: { eq: "trex.png" }) {
+        childImageSharp {
+          fluid(maxWidth: 200, grayscale: true) {
+            ...GatsbyImageSharpFixed_withWebp_noBase64
+          }
+        }
+      }
+    }
+  `)
+
+  return <Img fixed={data?.file?.childImageSharp?.fixed} alt="T-Rex" />
+}
+```
+
+## How does it work?
+
+When your site is compiled, any references to StaticImage components are extracted, the images are resized by Sharp in a similar way to `gatsby-transformer-sharp`, and then the resulting sharp object is written to `.cache/caches/gatsby-plugin-static-image/`, with the filename generated as a hash of the normalized image props. Next, a Babel plugin finds any references to StaticImage, calculates the same hash, loads the JSON file with the sharp object, then adds it as a new `parsedValues` prop. It then returns a GatsbyImage, passing the parsedValues as the fixed or fluid prop. Errors don't cause the build to fail, but instead are written to the component as an `__error` prop, which is then logged in develop.
+
+### Are there restrictions to how this is used?
+
+The props must be able to be statically-analyzed at build time. You can't pass them as props from outside the component, or use the results of function calls, for example.
+
+```js
+//Doesn't work
+({ logo }) => <Img src={logo}>
+```
+
+...and nor does this:
+
+```js
+//Doesn't work
+() => {
+    const width = getTheWidthFromSomewhere();
+    return <Img src="trex-png" width={width}>
+}
+```
+
+You can use variables and expressions if they're in the scope of the component, e.g.:
+
+```js
+//OK
+() => {
+    const width = 300
+    return <Img src="trex-png" width={width}>
+}
+```
+
+```js
+//Also OK
+() => {
+    const width = 300
+    const height = width * 16 / 9
+    return <Img src="trex-png" width={width} height={height}>
+}
+```
+
+## Installation
+
+```bash
+npm install gatsby-plugin-static-image
+```
+
+...then add it to your `gatsby-config.js`:
+
+```js
+module.exports = {
+  //...
+  plugins: [
+    "gatsby-plugin-sharp",
+    "gatsby-plugin-static-image",
+    //...
+  ],
+}
+```
+
+### API
+
+The only required prop is `src`. The default type is `fixed`.
+
+For now:
+
+```typescript
+export interface ImageOptions {
+  fixed?: boolean //Default true
+  fluid?: boolean
+  webP?: boolean
+  base64?: boolean // Default true
+  tracedSVG?: boolean
+}
+
+export interface SomeGatsbyImageProps {
```

Comment:
❓ I'm feeling that this owns a lot of props from the`img` tag. Are there any particular reasons not to extend `React.HTMLAttributes<HTMLImageElement>` ?

Reply:
It's passing them to gatsby-image, so we need to follow the gatsby-image props. I've split those props up though so that we can take the ones we need for the resizing etc too.